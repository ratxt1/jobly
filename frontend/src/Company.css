@import url("https://use.fontawesome.com/releases/v5.3.1/css/all.css");

.loading {
  width: 4em;
  margin: 200px auto 0 auto;
  color: lightskyblue;
}