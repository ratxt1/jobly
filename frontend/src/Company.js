import React, { Component } from 'react';
import './Company.css';
import JoblyApi from './JoblyApi'
import JobCard from './JobCard'


class Company extends Component {
  constructor(props) {
    super(props)
    this.state = {
      companyData: {},
      loading: true
    }
    this.applyToJob = this.applyToJob.bind(this)
  }

  /** update job application statuses for current user */
  async componentDidMount() {
    try {
      let companyData = await JoblyApi.getCompany(this.props.companyName)
      // get ids for jobs that curr user has applied for
      let currUserJobIds = this.props.currUser.jobs.map(job => job.id);
      // update application statuses for those jobs 
      companyData.jobs = companyData.jobs.map(job => {
        return currUserJobIds.includes(job.id)
          ? { ...job, state: "applied" }
          : job
      })
      this.setState({
        companyData,
        loading: false
      })
    } catch (err) {
      console.error("We got an error: ", err);
    }

  }

  /** given job id, make API call to apply to job & set state
   * NOTE: passed down to JobCard as a prop
   */
  async applyToJob(id) {
    let appliedJobStatus = await JoblyApi.applyToJob(id, "applied");
    this.setState(st => ({
      companyData: {
        ...st.companyData, 
        jobs: st.companyData.jobs.map(function (job) {
          return job.id === +id
            ? { ...job, state: appliedJobStatus }
            : job
        })
      }
    }))
  }

  render() {
    if (this.state.loading) {
      return (
        <div className="loading">
          <i className="fas fa-4x fa-spinner fa-spin" />
        </div>
      );
    }

    let { name, description, jobs } = this.state.companyData;

    let companyJobs = jobs.map(j => <JobCard
      key={j.id}
      id={j.id}
      title={j.title}
      salary={j.salary}
      equity={j.equity}
      state={j.state}
      apply={this.applyToJob}
    />);

    return (
      <div className="Company row">
        <div className="col-8 offset-2">
          <h2>{name}</h2>
          <p>{description}</p>
          <div className="CardList">
            {companyJobs}
          </div>
        </div>
      </div>
    )
  }
}

export default Company;
