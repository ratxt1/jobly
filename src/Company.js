import React, { Component } from 'react';
import './Company.css';
import JoblyApi from './JoblyApi'
import JobCard from './JobCard'


class Company extends Component{
    constructor(props) {
        super(props)
        this.state = {
            companyData:{},
            loading: true
        }
    }

    async componentDidMount() {
        let companyData = await JoblyApi.getCompany(this.props.companyName)
        this.setState({ 
            companyData: companyData,
            loading: false
        })
    }


    render(){
        if(this.state.loading){
            return (
                <div className="loading">
                <i className="fas fa-4x fa-spinner fa-spin" />
                </div>
            );
    }
    
    let { name, description, jobs } = this.state.companyData; 
    let companyJobs = jobs.map(j => <JobCard key={j.id} {...j}/>);

    return(
      <div className="Company">
        <h2>{ name }</h2>
        <p>{ description }</p>
        <div className="CardList">
            { companyJobs }
        </div>
      </div>      
    )
  }
}

export default Company;
