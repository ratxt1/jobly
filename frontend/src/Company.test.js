import React from "react";
import { shallow, mount } from "enzyme";
import { MemoryRouter } from 'react-router-dom';
import toJson from "enzyme-to-json";
import Company from "./Company";

let wrapper;


it("renders without crashing", function() {
    shallow(<Company />);
  });
  
// snapshot test
it("matches snapshot", function() {
    let wrapper = shallow(<Company />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});

it("runs componentDidMount", function(){
    const mockComponentDidMount = jest.spyOn(
        Company.prototype,
        "componentDidMount"
    )
    wrapper = mount(<Company />);
    expect(mockComponentDidMount).toHaveBeenCalled();
})