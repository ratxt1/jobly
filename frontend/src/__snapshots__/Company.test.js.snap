// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="loading"
>
  <i
    className="fas fa-4x fa-spinner fa-spin"
  />
</div>
`;
